src/hmem_cuda.c

```diff
@@ -610,10 +611,10 @@ int cuda_hmem_init(void)
 	if (ret != FI_SUCCESS)
 		goto dl_cleanup;
 
-	ret = 1;
+	param_value = 1;
 	fi_param_get_bool(NULL, "hmem_cuda_use_gdrcopy",
-			  &ret);
-	hmem_cuda_use_gdrcopy = (ret != 0);
+			  &param_value);
```

Comment:
You can check the result from fi_param_get to know if the variable was set by the app (ret == 0), and update the value accordingly.  E.g.
```
ret = fi_param_get(..)
if (!ret)
    hmem_cuda_use_gdr_copy = param_value ? XFER_ENABLED : XFER_DISABLED
```

Reply:
This gdrcopy change is out of scope for this PR. I removed this change in new revision.